src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp

```diff
@@ -1043,6 +1055,80 @@ struct npc_darkfallen_tactician : public DarkFallenAI
     }
 };
 
+struct npc_nerubar_broodkeeper : public ScriptedAI
+{
+    npc_nerubar_broodkeeper(Creature* creature) : ScriptedAI(creature)
+    {
+        Initialize();
+    }
+
+    void Initialize()
+    {
+        CastNet = false;
+    }
+
+    void Reset() override
+    {
+        events.Reset();
+        events.ScheduleEvent(EVENT_CRYPT_SCARABS, 1s, 3s);  // Crypt Scarabs
+        events.ScheduleEvent(EVENT_DARK_MENDING, 10s, 15s); // Dark Mending
+        events.ScheduleEvent(EVENT_WEB_WRAP, 6s, 8s);       // Web Wrap
+    }
+
+    void SetData(uint32 type, uint32 data) override
+    {
+        if (type == 1 && data == 1)
+        {
+            me->GetMotionMaster()->MoveFall();
+            me->SetHover(false);
+            me->SetByteValue(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_ANIM_TIER, 0);
+            if (!CastNet)
+            {
+                CastNet = true;
+                me->RemoveAllAuras();
+                DoCastSelf(SPELL_CAST_NET_STARTROOM, true);
```

Comment:
You can make it simple and move everything here to **OnlyOnce** areatrigger script and restore SAI because so far I see no reason to script that creature in core

Reply:
Okey